presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -493,6 +497,24 @@ private synchronized void updatePools(Map<MemoryPoolId, Integer> queryCounts)
         }
     }
 
+    public synchronized Map<String, Optional<MemoryInfo>> getWorkerMemoryInfo()
+    {
+        Map<String, Optional<MemoryInfo>> memoryInfo = new HashMap<>();
+        AllNodes allNodes = nodeManager.getAllNodes();
+        Set<Node> workers = union(union(allNodes.getActiveNodes(), allNodes.getInactiveNodes()), allNodes.getShuttingDownNodes());
+        for (Entry<String, RemoteNodeMemory> entry : nodes.entrySet()) {
+            Optional<Node> worker = workers.stream()
+                    .filter(node -> node.getNodeIdentifier().equals(entry.getKey()))
+                    .findFirst();
+            String workerId = entry.getKey();
+            if (worker.isPresent()) {
+                workerId += " [" + worker.get().getHostAndPort().getHostText() + "]";
```

Comment:
Why not add `Node` to `RemoteNodeMemory`?

Reply:
Yep, good idea.